src/core/stream/index.ts

```diff
@@ -194,31 +199,47 @@ export default function Stream({
     openMediaSource(videoElement),
     fetchManifest(url)
   ).pipe(mergeMap(([ mediaSource, manifest ]) => {
+    const loadStream = StreamLoader({ // Behold!
+      mediaElement: videoElement,
+      manifest,
+      clock$,
+      speed$,
+      abrManager,
+      segmentPipelinesManager,
+      refreshManifest: fetchManifest,
+      bufferOptions: objectAssign({
+        textTrackOptions,
+        offlineRetry: networkConfig.offlineRetry,
+        segmentRetry: networkConfig.segmentRetry,
+      }, bufferOptions),
+    });
 
     log.debug("calculating initial time");
     const initialTime = getInitialTime(manifest, startAt);
     log.debug("initial time calculated:", initialTime);
 
-    return observableConcat(
-      observableOf(EVENTS.manifestReady(abrManager, manifest)),
-      loadStreamOnMediaSource({ // Behold!
-        mediaElement: videoElement,
-        mediaSource,
-        manifest,
-        initialSettings: { time: initialTime, shouldPlay: autoPlay },
-        clock$,
-        speed$,
-        abrManager,
-        segmentPipelinesManager,
-        refreshManifest: fetchManifest,
-        bufferOptions: objectAssign({
-          textTrackOptions,
-          offlineRetry: networkConfig.offlineRetry,
-          segmentRetry: networkConfig.segmentRetry,
-        }, bufferOptions),
+    const reloadStreamSubject$ = new Subject<void>();
+    (window as any).reload = reloadStreamSubject$;
+    const reloadStream$ : Observable<IStreamEvent> = reloadStreamSubject$.pipe(
+      switchMap(() => {
+        const currentPosition = videoElement.currentTime;
+        const isPaused = videoElement.paused;
+        return openMediaSource(videoElement).pipe(
+          mergeMap(newMS => loadStream(newMS, currentPosition, !isPaused)),
+          startWith(EVENTS.reloadingStream())
+        );
       })
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=284&issues=AWU38rlxOTG07BqbHTWj&open=AWU38rlxOTG07BqbHTWj)

[ISSUE_KEY]: AWU38rlxOTG07BqbHTWj